src/hera/workflows/script.py

```diff
@@ -268,13 +238,105 @@ def script_wrapper(func: Callable) -> Callable:
         """
         s = Script(name=func.__name__.replace("_", "-"), source=func, **script_kwargs)
 
-        def task_wrapper(**task_params) -> Union[Task, Step]:
+        def task_wrapper(**kwargs) -> Union[Task, Step]:
             """Invokes a `Script` object's `__call__` method using the given `task_params`"""
-            return s.__call__(**task_params)  # type: ignore
+            if _context.active:
+                return s.__call__(**kwargs)  # type: ignore
+            return func(**kwargs)
 
+        # Set the wrapped function to the original function so that we can use it later
+        task_wrapper.wrapped_function = func  # type: ignore
```

Comment:
We set the original function as an attribute to allow hera runner to correctly call the original function regardless of whether it was wrapped or not.